csss-site/src/tools/lib/generate_document.py

```diff
@@ -0,0 +1,222 @@
+
+    payable_to: str
+    amount_cad: float
+    receive_kind: typing.Union[Pickup, Mailed]
+
+    current_date: datetime
+
+
+@dataclass
+class ReceiptInfo:
+    paid_by: str
+    location: str
+    amount_cad: float
+
+# ------------------------
+# functions
+
+
+# generate cheque req from core
+# meeting_minutes: str is the prefix for the meeting minutes of the form: "yyyy-mm-dd"
+def generate_core_cheque_req(
+    data: CoreChequeReq,
+        meeting_minutes: str,
+        receipts: list[ReceiptInfo],
+        include_auto_desc: bool = True,
+        watermark_active: bool = True):
+
+    reader = PdfReader(cheque_req_form_pickup)
+    assert len(reader.pages) == 1
+
+    page = reader.pages[0]
+
+    # TODO: automatically generate a description given info about receipts
+    # TODO: automatically include attached meeting minutes in the cheque-req
+    #       -> do a search; look for person's name & reciept costs
+    # TODO: in the web tool, have a link to this script's source, and an existing filled out form
+    #       (or something like that)
+
+    relevant_fields = {
+        'Today s Date': data.current_date.strftime("%B %#d, %Y"),
+        'Cheque Payable To print legibly': data.payable_to,
+        'In The Amount Of': "${0:.2f}".format(data.amount_cad),
+        'Describe the request andor provide additional information if necessary': data.request_desc,
+
+        'Requested By': data.request_by,
+        'Position': data.requester_position,
+    }
+
+    # contextual fields
+    if isinstance(data.receive_kind, Pickup):
+        relevant_fields['Picked up by'] = data.receive_kind.pickup_by
+        relevant_fields['Email'] = data.receive_kind.pickup_email
+    elif isinstance(data.receive_kind, Mailed):
+        relevant_fields['Street Address'] = data.receive_kind.street_address
+        relevant_fields['City, Province'] = data.receive_kind.city_province
+        relevant_fields['Postal Code'] = data.receive_kind.postal_code
+
+        # check boxes are not easy, so just update them in the reader's page
+        found_check_box_1 = False
+        found_check_box_3 = False
+        if "/Annots" in page:
+            for annotation in page["/Annots"]:
+                annotation_obj = annotation.get_object()
+                annotation_type = annotation_obj['/T']
+                if annotation_type == "Check Box1":
+                    found_check_box_1 = True
+                elif annotation_type == "Check Box3":
+                    found_check_box_3 = True
+
+                if annotation_type == "Check Box1" and data.receive_kind.mail_off_campus or \
+                   annotation_type == "Check Box3" and not data.receive_kind.mail_off_campus:
+                    annotation_obj.update({
+                        NameObject("/V"): NameObject("/Yes"),
+                        NameObject("/AS"): NameObject("/Yes")
+                    })
+        else:
+            raise KeyError("pdf has no checkboxes; something is wrong!")

```

Comment:
one problem with raising an error is it completely reset the user's view and they lose their place, is that Ok for this?